src/components/InfoPanel/InfoPanel.js

```diff
@@ -49,12 +49,25 @@ const InfoPanel = () => {
 
   const sendAnalytics = useAnalytics();
 
+  // Close topology, if open, on Escape key press
+  useEffect(() => {
+    const closeOnEscape = function (e) {
+      if (e.key === "Escape" && showExpandedTopology) {
+        setShowExpandedTopology(false);
+      }
+    };
+    document.addEventListener("keydown", closeOnEscape);
+    return () => {
+      document.removeEventListener("keydown", closeOnEscape);
+    };
+  });
+
   return (
     <div className="info-panel">
       {showExpandedTopology ? (
         <Modal
           close={() => setShowExpandedTopology(false)}
-          title={modelName.split("/")[1]}
+          title={modelName.split("/")[1] || "Error: model not loading"}
```

Comment:
This looks good but maybe we should also put a conditional around the expand button so that the modal can't even be seen if we don't have the model data.